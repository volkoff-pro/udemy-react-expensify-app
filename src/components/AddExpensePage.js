import React from 'react';
import { connect } from 'react-redux';
import { func, object, shape } from 'prop-types';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses.actions';

const AddExpensePage = ({ dispatch, history }) => (
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      onSubmit={expense => {
        dispatch(addExpense(expense));
        history.push('/');
      }}
    />
  </div>
);

AddExpensePage.defaultProps = {
  history: undefined
};

AddExpensePage.propTypes = {
  dispatch: func.isRequired,
  history: shape(object.isRequired)
};

export default connect()(AddExpensePage);
