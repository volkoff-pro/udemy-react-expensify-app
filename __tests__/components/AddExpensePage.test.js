import React from 'react';
import { shallow, mount, render } from 'enzyme';
import AddExpensePage from '../../src/components/AddExpensePage';

describe('AddExpensePage component', () => {
  test('should shallow correctly', () => {
    expect(shallow(<AddExpensePage />)).toMatchSnapshot();
  });
  test('should mount correctly', () => {
    expect(mount(<AddExpensePage />)).toMatchSnapshot();
  });
  test('should render correctly', () => {
    expect(render(<AddExpensePage />)).toMatchSnapshot();
  });
});
