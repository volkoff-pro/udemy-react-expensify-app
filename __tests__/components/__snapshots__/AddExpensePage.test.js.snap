// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpensePage component should mount correctly 1`] = `
<AddExpensePage>
  <div>
    This is from my add expense component
  </div>
</AddExpensePage>
`;

exports[`AddExpensePage component should render correctly 1`] = `
<div>
  This is from my add expense component
</div>
`;

exports[`AddExpensePage component should shallow correctly 1`] = `
<div>
  This is from my add expense component
</div>
`;
